### PR TITLE
Action bar refactor

### PR DESCRIPTION
**Pull request by lee-chase**
Contributes to #488 

Refactor ActionBar away from using child 

#### What did you change?

- ActionBar to use actions array property instead of children.
- Deprecated use of children

#### How did you test and verify your work?

Ran tests and viewed in Storybook.

### CODE REVIEW

File: packages/cloud-cognitive/src/components/ActionBar/ActionBar.stories.js [hunk 1]
```diff
@@ -8,40 +8,45 @@
...
+  onClick: action(`Action ${num}`),
+}));
+
 const actionBarItems = (
   <>
     <ActionBarItem
       renderIcon={Lightning16}
       iconDescription="Action 1"
       onClick={action('Action 1')}
+      data-wibble="wibble"
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/ActionBar/ActionBar.stories.js:49]: Though I always approve of wibble in all its forms, does this line belong here?

**Review by dcwarwick — CHANGES_REQUESTED** [review]

File: packages/cloud-cognitive/src/components/ActionBar/ActionBar.js [hunk 2]
```diff
@@ -195,21 +205,38 @@ export let ActionBar = React.forwardRef(
...
    * className
    */
   className: PropTypes.string,
   /**
    * maximum visible ActionBarItems
    */
-  maxVisibleActionBarItems: PropTypes.number,
+  maxVisible: PropTypes.number,
   /**
    * onItemCountChange - event reporting maxWidth
```
[the remark this answers is not in the document]
Comment (lee-chase) [reply packages/cloud-cognitive/src/components/ActionBar/ActionBar.js:241]: Updated

File: packages/cloud-cognitive/src/components/ActionBar/ActionBar.stories.js (hunk 1 above)
Comment (lee-chase) [reply packages/cloud-cognitive/src/components/ActionBar/ActionBar.stories.js]: Removed

File: packages/cloud-cognitive/src/components/ActionBar/ActionBar.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dcwarwick) [reply packages/cloud-cognitive/src/components/ActionBar/ActionBar.js:241]: Not actually seeing a change here? the comment says "onItemCountChange" while the prop is "onWidthChange", and I'm not sure what the reference to 'maxWidth' means either. But it's not really part of this PR which I'm happy to approve :-) The doc comments will need a little bit of work. I'm not sure that it's useful for a docgen comment to start by repeating the name of the prop it's describing.

**Review by dcwarwick — APPROVED** [review]

**Pull request merged**